v2/kafka-to-bigquery/src/main/java/com/google/cloud/teleport/v2/transforms/AvroTransform.java

```diff
@@ -125,12 +171,17 @@ private static class KafkaRecordToGenericRecordFailsafeElementFn
     private Schema schema = null;
     private String topicName;
     private String useConfluentWireFormat;
+    private BadRecordRouter badRecordRouter;
 
     KafkaRecordToGenericRecordFailsafeElementFn(
-        Schema schema, String topicName, String useConfluentWireFormat) {
+        Schema schema,
+        String topicName,
+        String useConfluentWireFormat,
+        BadRecordRouter badRecordRouter) {
       this.schema = schema;
       this.topicName = topicName;
       this.useConfluentWireFormat = useConfluentWireFormat;
+      this.badRecordRouter = badRecordRouter;
```

Comment:
If these fields are only set in the constructor, declare them final.

Reply:
getting errors when declaring them final